File: snake_water_gun.py
import random
def comparision(comp,user_choice):
    if(comp==user_choice):
        return None
    elif(comp=='s'):
        if(user_choice=='g'):
            return True
        elif(user_choice=='w'):
            return False
    elif(comp=='w'):
        if(user_choice=='s'):
            return True
        elif(user_choice=='g'):
            return False
    elif(comp=='g'):
        if(user_choice=='w'):
            return True
        elif(user_choice=='s'):
            return False

print("\n\t\t\t\tcomputer Turn:  snake(s),gun(g) and water(w) ") 
comp_choice=random.randint(1,100)
if 0< comp_choice <30:
    comp='s'
elif 30 < comp_choice <60:
    comp='w'
else:
    comp='g'

user_choice=input("\n\t\t\t\t Your Turn: snake(s),gun(g) and water(w) :")

result=comparision(comp,user_choice)

if result==None:
    print("\t\t\t\t--------------------------------------------------------------")
    print("\n\t\t\t\tMatch Tie : Both you and computer chose same event")
    print("\t\t\t\t--------------------------------------------------------------")
elif result==True:
    print("\t\t\t\t--------------------------------------------------------------")
    print('''\n\t\t\t\tHurrey YOU Win The Match
              \t\t\t\tEnjoy your day!!!''')
    print("\t\t\t\t--------------------------------------------------------------")

elif result==False:
    print("\t\t\t\t--------------------------------------------------------------")
    print('''\n\t\t\t\tOhh No !! You Lose The Match 
              \t\t\t\t    Try Next Time 
            \t\t\t\t   Good Day!!''')
    print("\t\t\t\t--------------------------------------------------------------")

File: snake_water_gun.cpp
#include <iostream>
#include <time.h>

using namespace std;
//C ++ language have  bool data type so  return value in true and false;
int comparision(char comp, char user_choice)
{
    if (comp == user_choice)
        return 0;

    else if (comp == 's')
    {
        if (user_choice == 'w')
            return -1;
        else if (user_choice == 'g')
            return 1;
    }
    else if (comp == 'w')
    {
        if (user_choice == 's')
            return 1;
        else if (user_choice == 'g')
            return -1;
    }
    if (comp == 'g')
    {
        if (user_choice == 'w')
            return 1;
        else if (user_choice == 's')
            return -1;
    }

    return 0; //warning: control reaches end of non-void function  (so i have to add this to non void function)
}

int main()
{

    char comp;
    char user_choice;
    int comp_choice;

    cout << "\n\t\t\t\t Computer Turn : snake(s),gun(g) and water(w)\n"
         << endl;

    srand(time(0));
    comp_choice = rand() % 100 + 1;
    // cout << comp_choice;
    if (comp_choice < 33)
    {
        comp = 's';
    }
    else if (comp_choice > 33 && comp_choice < 66)
    {
        comp = 'w';
    }
    else
    {
        comp = 'g';
    }

    // cout << comp;
    cout << "\n\t\t\t\tYour Turn : snake(s), gun(g) and water(w) :";
    cin >> user_choice;

    int result = comparision(comp, user_choice);

    if (result == 0)
    {
        cout << "\t\t\t\t--------------------------------------------------------------" << endl;
        cout << "\n\t\t\t\tMatch Tie : Both you and computer chose same event\n" << endl;
        cout << "\t\t\t\t--------------------------------------------------------------" << endl;
    }
    else if (result == 1)
    {
        cout << "\t\t\t\t--------------------------------------------------------------" << endl;
        cout << "\n\t\t\t\tHurrey YOU Win The Match\n\t\t\t\tEnjoy your day!!!\n"
             << endl;
        cout << "\t\t\t\t--------------------------------------------------------------" << endl;
    }
    else if (result == -1)
    {
        cout << "\t\t\t\t--------------------------------------------------------------" << endl;
        cout << "\n\t\t\t\tOhh No !! You Lose The Match \n\t\t\t\t    Try Next Time \n\t\t\t\t   Good Day!!\n"
             << endl;
        cout << "\t\t\t\t--------------------------------------------------------------" << endl;
    }
    return 0;
}